File: frontend/fix-imports.js
const fs = require('fs');
const path = require('path');

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(file => {
        file = path.join(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) {
            results = results.concat(walk(file));
        } else if (file.endsWith('.tsx') || file.endsWith('.ts')) {
            results.push(file);
        }
    });
    return results;
}

const files = walk('./src');
const replacements = ['Button', 'Card', 'Input', 'Popover', 'Calendar', 'Checkbox', 'Label', 'Skeleton', 'Slider', 'Separator', 'Badge'];

let changedFiles = 0;
files.forEach(file => {
    let content = fs.readFileSync(file, 'utf8');
    let changed = false;
    
    replacements.forEach(r => {
        // match ui/Button or ui/Card
        const regex = new RegExp('ui/' + r + '(?=[\'\"])', 'g');
        if (regex.test(content)) {
            content = content.replace(regex, 'ui/' + r.toLowerCase());
            changed = true;
        }
        
        // match ./Button or ./Card inside components/ui
        const regex2 = new RegExp('\\./' + r + '(?=[\'\"])', 'g');
        if (file.includes('components\\\\ui') || file.includes('components/ui')) {
            if (regex2.test(content)) {
                content = content.replace(regex2, './' + r.toLowerCase());
                changed = true;
            }
        }
    });
    
    if (changed) {
        fs.writeFileSync(file, content);
        changedFiles++;
    }
});
console.log('Changed ' + changedFiles + ' files');

File: frontend/fix-imports.py
import os
import re

replacements = ['Button', 'Card', 'Input', 'Popover', 'Calendar', 'Checkbox', 'Label', 'Skeleton', 'Slider', 'Separator', 'Badge']

for root, _, files in os.walk('src'):
    for file in files:
        if file.endswith('.tsx') or file.endswith('.ts'):
            path = os.path.join(root, file)
            with open(path, 'r', encoding='utf-8') as f:
                content = f.read()
            original = content
            for r in replacements:
                # Replace ui/Card with ui/card
                content = re.sub(r'ui/' + r + r'(?=[\'\"])', 'ui/' + r.lower(), content)
                # Replace ./Card with ./card inside components/ui
                if 'components\\\\ui' in path or 'components/ui' in path:
                    content = re.sub(r'\./' + r + r'(?=[\'\"])', './' + r.lower(), content)
            if content != original:
                with open(path, 'w', encoding='utf-8') as f:
                    f.write(content)
                print(f'Fixed {path}')
